### PR TITLE
[feature](nereids) adjust min/max of column stats for cast function

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
1. cast(A as date), where A is a string column. the min/max of result column stats should be calc like this:
convert A.minExpr to a date dateA, and then get double value from dateA.

2. add "explain memo plan select ..." to print memo from mysql client

3. dump column stats for FileScanNode, used in datalake.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java [hunk 1]
```diff
@@ -314,8 +314,8 @@ public boolean enclosed(ColumnStatistic other) {
 
     @Override
     public String toString() {
-        return isUnKnown ? "unKnown" : String.format("ndv=%.4f, min=%f, max=%f, sel=%f, count=%.4f",
-                ndv, minValue, maxValue, selectivity, count);
+        return isUnKnown ? "unKnown" : String.format("ndv=%.4f, min=%f(%s), max=%f(%s), count=%.4f",
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java:317]:
```suggestion
        return isUnKnown ? "unknown" : String.format("ndv=%.4f, min=%f(%s), max=%f(%s), count=%.4f",
```

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.26 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          33 seconds loaded 861443392 Bytes, about 24 MB/s
 insert into select:          33.4 seconds inserted 10000000 Rows, about 299K ops/s
 storage size: 17162493751 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713010233_clickbench_pr_177539.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/ExpressionEstimation.java [hunk 2]
```diff
@@ -140,12 +143,38 @@ public ColumnStatistic visitCaseWhen(CaseWhen caseWhen, Statistics context) {
...
             return stats;
         }
-        return cast.child().accept(this, context);
+        ColumnStatistic childColStats = cast.child().accept(this, context);
+
+        return castMinMax(childColStats, cast.getDataType());
+    }
+
+    private ColumnStatistic castMinMax(ColumnStatistic colStats, DataType targetType) {
+        if (colStats.minExpr instanceof StringLiteral && targetType.isDateLikeType()) {
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/ExpressionEstimation.java:158]: yes. this function should be expanded!

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java [hunk 3]
```diff
@@ -737,8 +738,9 @@ public String toString() {
             builder.append("\n\n").append(group);
             builder.append("  stats=").append(group.getStatistics()).append("\n");
             Statistics stats = group.getStatistics();
+            Plan plan = group.getLogicalExpressions().get(0).getPlan();
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java:741]: done

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java]: thx!

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run beut feut clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.42 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         18 seconds loaded 2358488459 Bytes, about 124 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17160485797 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230714115452_clickbench_pr_178322.html

**Pull request merged**